Example-Swift/PhotosProvider.swift

```diff
@@ -7,39 +7,43 @@
 //
 
 import UIKit
-import NYTPhotoViewer
 
-/**
-*   In Swift 1.2, the following file level constants can be moved inside the class for better encapsulation
-*/
-let CustomEverythingPhotoIndex = 1, DefaultLoadingSpinnerPhotoIndex = 3, NoReferenceViewPhotoIndex = 4
-let PrimaryImageName = "NYTimesBuilding"
-let PlaceholderImageName = "NYTimesBuildingPlaceholder"
+/// A component of your data layer, which might load photos from the cache or network.
+final class PhotosProvider {
+    typealias Slideshow = [Photo]
 
-class PhotosProvider: NSObject {
+    /// Simulate a synchronous fetch of a slideshow, perhaps from a local database.
+    func fetchDemoSlideshow() -> Slideshow {
+        return Array(0...4).map { demoPhoto(identifier: $0) }
```

Comment:
You don't need to wrap the range in an array initializer here; `(0...4).map { … }` will also do what you want!